File: ListViewExample/app/src/main/java/com/tlu/listviewexample/MyAdapter.java
package com.tlu.listviewexample;
import com.tlu.listviewexample.model.Category;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;



import java.util.ArrayList;


public class MyAdapter extends ArrayAdapter<Category> {
    public MyAdapter(@NonNull Context context, ArrayList<Category> lst) {
        super(context, 0, lst);

    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {

        View currentItemView = convertView;
        if (currentItemView == null) {
            currentItemView = LayoutInflater.from(getContext()).inflate(R.layout.item_list_view, parent, false);

        }
        Category category = getItem(position);
        ImageView imageView = (ImageView) currentItemView.findViewById(R.id.img_product);
        TextView txtView = (TextView) currentItemView.findViewById(R.id.txt_title);


        assert category != null;
        imageView.setImageResource(category.getImg());

        txtView.setText(category.getTitle());

        return currentItemView;


    }
}

File: FireBaseExample/app/src/main/java/com/tlu/firebaseexample/MyAdapter.kt
package com.tlu.firebaseexample

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

class MyAdapter(context: Context, objects: ArrayList<Student>) :
    ArrayAdapter<Student>(context, 0, objects) {
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val view: View = convertView ?: LayoutInflater.from(context)
            .inflate(R.layout.custom_list_view, parent, false)


        val st = getItem(position)

        val txtView = view.findViewById<TextView>(R.id.textView)
        txtView.setText("$position - ${st?.name} - ${st?.email}")

        return view
    }
}